clang/test/CodeGenSYCL/code_location.cpp

```diff
@@ -0,0 +1,101 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem -sycl-std=2020 -fsycl-int-header=%t.h %s -o %t.out
+// RUN: FileCheck -input-file=%t.h %s
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem -sycl-std=2020 -DNDEBUG -fsycl-int-header=%t2.h %s -o %t2.out
+// RUN: FileCheck -input-file=%t2.h %s
```

Comment:
I think it would help if a host compilation that consumes the generated header file is also added.

Reply:
@premanandrao is this what you mean?